CMakeLists.txt

```diff
@@ -156,6 +157,19 @@ mark_as_advanced(ENABLE_SENTRY)
 cmake_dependent_option(FORCE_LEGACY_LIBBPF "Force usage of libbpf 0.0.9 instead of the latest version." False "ENABLE_PLUGIN_LIBBPF" False)
 mark_as_advanced(FORCE_LEGACY_LIBBPF)
 
+if(ENABLE_WOLFSSL)
+        set(ENABLE_OPENSSL False)
+        # two features are incompatible
+        set(ENABLE_PLUGIN_LOGS_MANAGEMENT False)
+        set(ENABLE_H2O False)
+        # OSSL_DECODER_CTX does not exist in WolfSSL
+        set(ENABLE_ACLK False)
+else()
+        set(ENABLE_OPENSSL True)
+        option(ENABLE_H2O "Enable H2O web server (experimental)" True)
+        mark_as_advanced(ENABLE_H2O)
+endif()
```

Comment:
Please do not use `set` on options or put options in conditional clauses. Instead, use https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html (we already include this module, so you should just need to modify the `option()` calls to use `cmake_dependent_option()` with the appropriate arguments).